.circleci/scripts/set-test-site-url.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+test_site_url=$(curl -H "Authorization: Bearer ${NETLIFY_ACCESS_TOKEN}" -H "User-Agent: CircleCI-${CIRCLE_PROJECT_USERNAME}-${CIRCLE_PROJECT_REPONAME}/${CIRCLE_BUILD_NUM}" "$NETLIFY_DEPLOY_PREVIEW_FUNCTION_URL"?commit="${CIRCLE_SHA1}"\&siteid="${NETLIFY_SITE_ID}" )
+echo "export TEST_SITE_URL=$test_site_url" >> "$BASH_ENV"
+# shellcheck source=/dev/null
+source "$BASH_ENV"
+echo "TEST_SITE_URL environment variable set: $test_site_url"
```

Comment:
The script should really check that we get a successful 200 response back, and abort the build if not.  That can come later - will be easier to add if/when we move this bash script to Go, for instance.